### PR TITLE
fix(combobox): address multiselectable state issues with new `TagGroup` component

### PR DESCRIPTION
**Pull request by jzempel**
## Description

Addresses some touchy expand/collapse state bugs for multiselectable tag groups. The `TagGroup` is an internal-only component that renders a list of `Tag` components and `{children}` (the `StyledTagsButton`) as a final sibling.

## Detail

Also shimmed in:
- an input `aria-labelledby` label readout hack for screenreaders
- preventing double readout of `isEditable={false}` combobox value

## Checklist

- [ ] :ok_hand: ~design updates will be Garden Designer approved (add the designer as a reviewer)~
- [x] :globe_with_meridians: demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :guardsman: includes new unit tests. Maintain [existing coverage](https://coveralls.io/github/zendeskgarden/react-components?branch=main) (always >= 96%)
- [x] :wheelchair: tested for [WCAG 2.1 AA](https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_customize&levels=aaa) accessibility compliance
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

**Review by geotrev — APPROVED** [review]
Nothing major here, just questions mainly. The interactions feel much better. Making the tags all visible on focus is a great workaround!

File: packages/dropdowns.next/src/elements/combobox/TagGroup.tsx
Comment (Francois-Esquire) [inline packages/dropdowns.next/src/elements/combobox/TagGroup.tsx:1]: [nit] Could we add `TagGroup` to the Storybook `subcomponents` list in the `Combobox` story?

**Review by Francois-Esquire — APPROVED** [review]

File: packages/dropdowns.next/src/elements/combobox/TagGroup.tsx
Comment (jzempel) [reply packages/dropdowns.next/src/elements/combobox/TagGroup.tsx:1]: No, it is an internal-only component.

File: packages/dropdowns.next/src/elements/combobox/Combobox.tsx [hunk 1]
```diff
@@ -265,7 +240,37 @@ export const Combobox = forwardRef<HTMLDivElement, IComboboxProps>(
...
+                        tabIndex={-1}
+                        type="button"
+                      >
+                        {(() => {
+                          const value = selection.length - maxTags;
+
+                          return renderExpandTags
+                            ? renderExpandTags(value)
+                            : expandTags?.replace('{{value}}', value.toString());
+                        })()}
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/dropdowns.next/src/elements/combobox/Combobox.tsx:270]: It's one (not the only) way I can cram the `const` declaration inline JSX. Not a super common pattern, but I really wasn't liking the feel of an extracted function combined with the linter complaining about needing `useCallback`. I'm open to other approaches.

**Pull request merged**